src/V3AstNodes.h

```diff
@@ -3817,15 +3820,18 @@ class AstCoverToggle final : public AstNodeStmt {
 class AstDelay final : public AstNodeStmt {
     // Delay statement
 public:
-    AstDelay(FileLine* fl, AstNode* lhsp)
+    AstDelay(FileLine* fl, AstNode* lhsp, AstNode* stmtsp)
         : ASTGEN_SUPER_Delay(fl) {
         setOp1p(lhsp);
+        setNOp2p(stmtsp);
     }
     ASTNODE_NODE_FUNCS(Delay)
     virtual bool same(const AstNode* samep) const override { return true; }
     //
-    AstNode* lhsp() const { return op1p(); }  // op2 = Statements to evaluate
+    AstNode* lhsp() const { return op1p(); }  // op1 = delay value
     void lhsp(AstNode* nodep) { setOp1p(nodep); }
+    void stmtsp(AstNode* nodep) { setOp2p(nodep); }  // op2 = statements under delay
+    AstNode* stmtsp() const { return op2p(); }
```

Comment:
Why do we need this here? It is very contra intuitive as the Delay itself applies to everything after it, and the only thing we seem to do with it is plonk if after the AstDelay in V3Width. If there is no good reason, then revert this to how it used to be otherwise add comment on why it's this way please

Reply:
I think I figured out this is for forks, still I would like a comment to confirm please

Reply:
Yes, it's for forks.

Also for a case like this:

```systemverilog
    initial @e #5 $finish;
```

With `--debug`, it causes an error:

```
%Error: Internal Error: test.sv:3:17: ../V3Ast.cpp:344: Adding list to non-list op2
-node: DELAY 0x5555560b1b80 <e21#> {c3ar}
    3 |     initial @e #5 $finish;
      |                 ^
```